compiler/damlc/daml-stdlib-src/DA/Internal/LF.daml

```diff
@@ -151,8 +151,9 @@ instance Show (ContractId a) where
     None -> "<contract-id>"
     Some t -> t
 
--- | HIDE This is currently used in the internals of Daml script and Daml triggers
--- but not really something that we want to expose to users.
+-- | Used to convert the type index of a `ContractId`, since they are just
+-- pointers. Note that subsequent fetches and exercises might fail if the
+-- template of the contract on the ledger doesn't match.
```

Comment:
Maybe we could add a recommendation to use more type-safe function, like `fromInterface`, `toInterface` , `fromRequiringInterface`